nautobot/extras/tests/integration/test_plugins.py

```diff
@@ -122,3 +122,25 @@ def test_plugin_webhook_with_body(self):
         with open(os.path.join(tempfile.gettempdir(), "test_plugin_webhook_with_body"), "r") as f:
             self.assertEqual(json.loads(f.read()), {"message": "created"})
         os.remove(os.path.join(tempfile.gettempdir(), "test_plugin_webhook_with_body"))
+
+
+class PluginReturnUrlTestCase(SeleniumTestCase):
+    """
+    Integration tests for the CustomField and CustomFieldChoice models.
+    """
+
+    def setUp(self):
+        super().setUp()
+        self.user.is_superuser = True
+        self.user.save()
+        self.login(self.user.username, self.password)
+
+    def test_plugin_return_url(self):
+        """This test ensures that plugins return url for new objects is the list view."""
+        self.browser.visit(f'{self.live_server_url}{reverse("plugins:example_plugin:examplemodel_add")}')
+
+        form = self.browser.find_by_tag("form")
+
+        # Check that the Cancel button takes you to the ExampleModel List view.
+        form.first.links.find_by_text("Cancel").click()
+        self.assertEqual(self.browser.find_by_tag("h1").value, "Example Models")
```

Comment:
Would it be simpler to check the `href` of the "Cancel" link and confirm that it matches `reverse("plugins:example_plugin:examplemodel_list")`?

Reply:
If I knew how to do that, then yes. I tried, but had a hard time ensuring I was getting the Cancel button...